### DISCUSSION TITLE
[Fix Suggestion] BezierWalkerWithTime may have a bug in the normalizedT calculations

### DISCUSSION THREAD

**Discussion by fgsebares** [yasirkula/UnityBezierSolution#33] [General]
Hello!

First of all, thank you a lot for sharing this amazing tool, I was having a headache trying to make Beziers to work, and this saved me hours or days of pain. So really thankful for it :)

I wanted to share what I think might be a bug. I'm using BezierWalkerWithTime, but no matter which time I inputed, it always went very fast, like 2 seconds or less. I dived into the code, and I saw this:

m_normalizedT += deltaTime / travelTime;

I think the intended operation was a direct assignment, not an addition. 

m_normalizedT = deltaTime / travelTime;

This way, it appears to work as intended.

I hope this helps!

**Comment by yasirkula** [comment]
Hi! Are you calling the Execute method manually? If so, when you pass Time.deltaTime as parameter, the current implementation should work correctly.

**Comment by michael-mackinnon** [comment]
@fgsebares: 
Assuming that you haven't changed the **Update()** method and are still calling **Execute()** in that **Update()** method:

**deltaTime** is the time for the current single frame. **travelTime** is the time to traverse the whole spline. 
So **deltaTime**/**travelTime** is the portion of the whole spline that the mesh should travel in this frame.
We need to add that to the **normalizedT** that were at before this frame to position ourselves for the current frame.

`normalizedT = deltaTime / travelTime;`
This will result in the mesh going to some part of the spline and, given constant frame times, staying there.

If the time that the mesh goes along the spline is too fast, I'd suggest reducing the _Travel Time_ even more. Making it a large value should slow it down a little. Change the _Travel Time_ to 30, or something like that. NOTE: You'll need a LARGE _Travel Time_ to make the mesh travel slowly.

Are you calling **Execute()** at some other time and passing in something other than **Time.deltaTime**? 
That would break the behaviour of the **BezierWalker**... That might break the expected behaviour of **BezierWalkerWithTime** and give you what you're seeing.

Not sure what you're doing, but as @yasirkula suggests, it's currently working as expected with +=. If it's not working in your case, there must be something else amiss.

Best of luck!
